marklogic-data-hub-central/ui/src/components/entities/matching/ruleset-single-modal/ruleset-single-modal.tsx

```diff
@@ -76,40 +100,68 @@ const MatchRulesetModal: React.FC<Props> = (props) => {
       case 'exact':
       case 'reduce':
       case 'zip':
-        let propertyName = selectedProperty || '';
-
-        let matchRule: MatchRule = {
-          entityPropertyPath: propertyName,
-          matchType: matchType,
-          options: {}
-        }
+        {
+          let propertyName = selectedProperty || '';
 
-        let matchRuleset: MatchRuleset = {
-          name: propertyName,
-          weight: 0,
-          matchRules: [matchRule]
+          let matchRule: MatchRule = {
+            entityPropertyPath: propertyName,
+            matchType: matchType,
+            options: {}
+          }
+  
+          let matchRuleset: MatchRuleset = {
+            name: propertyName,
+            weight: 0,
+            matchRules: [matchRule]
+          }
+  
+          if (propertyErrorMessage === '' && matchErrorMessage === '') {
+            // TODO save step to backend
+            let newStepArtifact: MatchingStep = curationOptions.activeStep.stepArtifact;
```

Comment:
These 5 lines are repeated below, but I assume that can be addressed when this is enhanced to talk to the backend.